docs/concepts/advanced/extensions.md

```diff
@@ -328,3 +329,286 @@ class MyComponent(Component):
+into a dictionary of arguments and options. These are then available
+as keyword arguments to the [`handle`](../../../reference/api#django_components.ComponentCommand.handle)
+method of the command.
+
+```python
+from django_components import CommandArg, ComponentCommand, ComponentsExtension
+
+class HelloCommand(ComponentCommand):
+    name = "hello"
+    help = "Say hello"
+
+    arguments = [
+        # Positional argument
+        CommandArg(
+            name_or_flags="name",
+            help="The name to say hello to",
+        ),
+        # Optional argument
+        CommandArg(
+            name_or_flags=["--shout", "-s"],
+            action="store_true",
+            help="Shout the hello",
+        ),
+    ]
+
+    def handle(self, name: str, *args, **kwargs):
+        shout = kwargs.get("shout", False)
+        msg = f"Hello, {name}!"
+        if shout:
+            msg = msg.upper()
+        print(msg)
+```
+
+You can run the command with arguments and options:
+
+```bash
+python manage.py components ext run my_ext hello John --shout
+>>> HELLO, JOHN!
+```
+
+!!! note
+
+    Command definitions are parsed with `argparse`, so you can use all the features of `argparse` to define your arguments and options.
+
+    See the [argparse documentation](https://docs.python.org/3/library/argparse.html) for more information.
+
+    django-components defines types as
+    [`CommandArg`](../../../reference/api#django_components.CommandArg),
+    [`CommandArgGroup`](../../../reference/api#django_components.CommandArgGroup),
+    [`CommandSubcommand`](../../../reference/api#django_components.CommandSubcommand),
+    and [`CommandParserInput`](../../../reference/api#django_components.CommandParserInput)
+    to help with type checking.
+
+!!! note
+
+    If a command doesn't have the [`handle`](../../../reference/api#django_components.ComponentCommand.handle)
+    method defined, the command will print a help message and exit.
+
+### Grouping Arguments
+
+Arguments can be grouped using [`CommandArgGroup`](../../../reference/api#django_components.CommandArgGroup)
+to provide better organization and help messages.
+
+Read more on [argparse argument groups](https://docs.python.org/3/library/argparse.html#argument-groups).
+
+```python
+from django_components import CommandArg, CommandArgGroup, ComponentCommand, ComponentsExtension
+
+class HelloCommand(ComponentCommand):
+    name = "hello"
+    help = "Say hello"
+
+    # Argument parsing is managed by `argparse`.
+    arguments = [
+        # Positional argument
+        CommandArg(
+            name_or_flags="name",
+            help="The name to say hello to",
+        ),
+        # Optional argument
+        CommandArg(
+            name_or_flags=["--shout", "-s"],
+            action="store_true",
+            help="Shout the hello",
+        ),
+        # When printing the command help message, `--bar` and `--baz`
+        # will be grouped under "group bar".
+        CommandArgGroup(
```

Comment:
These `CommandArg` and `CommandArgGroup` objects are passed to `argparse` functions. But those argparse functions don't provide a standalone typing for their input. So I created types like `CommandARg` and `CommandArgGroup`.